src/ai/backend/client/func/vfolder.py

```diff
@@ -117,102 +116,58 @@ async def rename(self, new_name):
 
     @api_function
     async def upload(self, files: Sequence[Union[str, Path]],
-                     basedir: Union[str, Path] = None,
-                     show_progress: bool = False):
-        base_path = (Path.cwd() if basedir is None
-                     else Path(basedir).resolve())
-        files = [Path(file).resolve() for file in files]
+                     basedir: Union[str, Path] = None):
+
+        base_path = (Path.cwd() if basedir is None else Path(basedir).resolve())
+
+        if basedir:
+            files = [basedir / Path(file) for file in files]
+        else:
+            files = [Path(file).resolve() for file in files]
+
         total_size = 0
         for file_path in files:
             total_size += Path(file_path).stat().st_size
-        tqdm_obj = tqdm(desc='Uploading files',
-                        unit='bytes', unit_scale=True,
-                        total=total_size,
-                        disable=not show_progress)
-        with tqdm_obj:
-            attachments = []
-            for file_path in files:
-                try:
-                    attachments.append(AttachedFile(
-                        str(Path(file_path).relative_to(base_path)),
-                        ProgressReportingReader(str(file_path),
-                                                tqdm_instance=tqdm_obj),
-                        'application/octet-stream',
-                    ))
-                except ValueError:
-                    msg = 'File "{0}" is outside of the base directory "{1}".' \
-                          .format(file_path, base_path)
-                    raise ValueError(msg) from None
 
+        session = api_session.get()
+        config = session.config
+        base_url = config.endpoint
+        session_create_url = base_url / 'folders/{}/create_upload_session' \
+                                        .format(self.name)
+        session_upload_url = base_url / "folders/_/tus/upload/"
```

Comment:
Just FYI: This will be changed to use response-provided URL after we implement the storage-proxy.